packages/tfchain_client/src/contracts.ts

```diff
@@ -294,23 +298,35 @@ class Contracts extends QueryContracts {
 
   @checkConnection
   async cancelService(options: CancelServiceOptions) {
-    const extrinsic = await this.client.api.tx.smartContractModule.serviceContractCancel(options.serviceId);
+    const extrinsic = this.client.api.tx.smartContractModule.serviceContractCancel(options.serviceId);
     return this.client.patchExtrinsic(extrinsic, { map: () => options.serviceId });
   }
 
   @checkConnection
   async setServiceFees(options: SetServiceFeesOptions) {
-    const extrinsic = await this.client.api.tx.smartContractModule.serviceContractSetFees(
+    const extrinsic = this.client.api.tx.smartContractModule.serviceContractSetFees(
       options.serviceId,
       options.baseFee,
       options.variableFee,
     );
     return this.client.patchExtrinsic<ServiceContract>(extrinsic);
   }
 
+  @checkConnection
+  async setExtraFees(options: SetDedicatedNodeExtraFeesOptions) {
+    const extrinsic = this.client.api.tx.smartContractModule.setDedicatedNodeExtraFee(options.nodeId, options.extraFee);
+    return this.client.patchExtrinsic<RentContract>(extrinsic);
+  }
+
+  @checkConnection
+  async getDedicatedNodePrice(options: GetDedicatedNodePriceOptions) {
+    const res = await this.client.api.query.smartContractModule.dedicatedNodesExtraFee(options.nodeId);
+    return res.toPrimitive() as number;
+  }

```

Comment:
the query methods should be in the query class at the top of this module

Reply:
done.